pkg/asset/installconfig/baremetal/host.go

```diff
@@ -0,0 +1,182 @@
+package baremetal
+
+import (
+	"strconv"
+
+	"github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/pkg/errors"
+	"gopkg.in/AlecAivazis/survey.v1"
+)
+
+// Host prompts the user for hardware details about a baremetal host.
+func Host() (*baremetal.Host, error) {
+	var host baremetal.Host
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Name",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.Name); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "BMC Address",
+				Help:    "The address for the BMC, e.g. ipmi://192.168.0.1",
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &host.BMC.Address); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "BMC Username",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.BMC.Username); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Password{
+				Message: "BMC Password",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.BMC.Password); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Boot MAC Address",
+			},
+			Validate: survey.ComposeValidators(survey.Required, macValidator),
+		},
+	}, &host.BootMACAddress); err != nil {
+		return nil, err
+	}
+
+	// Boot mode
+	// Fixme: Wait until #3952 lands to uncomment this block
+	/*
+		var bootMode v1alpha1.BootMode
+		survey.AskOne(&survey.Select{
+			Message: "Select Boot Mode:",
+			Options: []string{"UEFI", "Legacy"},
+		}, &bootMode, nil)
+	*/
+	rdHints := false
+	survey.AskOne(&survey.Confirm{
```

Comment:
how important are these device hints, are these always required ?
any prompts that are optional should be remove in favor of simpler / fewer questions.

If you get enough requests for adding certain features then we should add those case by case.

most advanced users should be moved towards
```
openshift-install create install-config
# edit the install config for more changes, or save it and bring it as-is later on
openshift-install create cluster
```

Reply:
It's going to be pretty common. I think asking whether they want to provide them or not was a good approach, but I'm happy to omit the questions entirely.

Reply:
Removed the root device hints in favor of editing install config it if necessary .